cmd/orb.go

```diff
@@ -464,45 +491,94 @@ func promoteOrb(opts orbOptions) error {
 		return fmt.Errorf("The version '%s' must be a dev version (the string should begin `dev:`)", version)
 	}
 
-	response, err := api.OrbPromote(context.Background(), opts.log, opts.cl, namespace, orb, version, segment)
+	response, err := api.OrbPromote(opts.apiOpts, namespace, orb, version, segment)
 	if err != nil {
 		return err
 	}
 
-	opts.log.Infof("Orb `%s` was promoted to `%s/%s@%s`.\n", ref, namespace, orb, response.HighestVersion)
-	opts.log.Info("Please note that this is an open orb and is world-readable.")
+	opts.apiOpts.Log.Infof("Orb `%s` was promoted to `%s/%s@%s`.\n", ref, namespace, orb, response.HighestVersion)
+	opts.apiOpts.Log.Info("Please note that this is an open orb and is world-readable.")
 	return nil
 }
 
 func createOrb(opts orbOptions) error {
 	var err error
-	ctx := context.Background()
 
 	namespace, orb, err := references.SplitIntoOrbAndNamespace(opts.args[0])
 
 	if err != nil {
 		return err
 	}
 
-	_, err = api.CreateOrb(ctx, opts.log, opts.cl, namespace, orb)
+	_, err = api.CreateOrb(opts.apiOpts, namespace, orb)
 
 	if err != nil {
 		return err
 	}
 
-	opts.log.Infof("Orb `%s` created.\n", opts.args[0])
-	opts.log.Info("Please note that any versions you publish of this orb are world-readable.\n")
-	opts.log.Infof("You can now register versions of `%s` using `circleci orb publish`.\n", opts.args[0])
+	opts.apiOpts.Log.Infof("Orb `%s` created.\n", opts.args[0])
+	opts.apiOpts.Log.Info("Please note that any versions you publish of this orb are world-readable.\n")
+	opts.apiOpts.Log.Infof("You can now register versions of `%s` using `circleci orb publish`.\n", opts.args[0])
 	return nil
 }
 
 func showSource(opts orbOptions) error {
 	ref := opts.args[0]
 
-	source, err := api.OrbSource(context.Background(), opts.log, opts.cl, ref)
+	source, err := api.OrbSource(opts.apiOpts, ref)
 	if err != nil {
 		return errors.Wrapf(err, "Failed to get source for '%s'", ref)
 	}
-	opts.log.Info(source)
+	opts.apiOpts.Log.Info(source)
+	return nil
+}
+
+func orbInfo(opts orbOptions) error {
+	ref := opts.args[0]
+	log := opts.apiOpts.Log
+
+	info, err := api.OrbInfo(opts.apiOpts, ref)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get info for '%s'", ref)
+	}
+
+	log.Info("\n")
+
+	if len(info.Orb.Versions) > 0 {
+		log.Infof("Latest: %s@%s", info.Orb.Name, info.Orb.HighestVersion)
+		log.Infof("Last-updated: %s", info.Orb.Versions[0].CreatedAt)
+		log.Infof("Created: %s", info.Orb.CreatedAt)
+		firstRelease := info.Orb.Versions[len(info.Orb.Versions)-1]
+		log.Infof("First-release: %s @ %s", firstRelease.Version, firstRelease.CreatedAt)
+
+		log.Infof("Total-revisions: %d", len(info.Orb.Versions))
+	} else {
+		log.Infof("This orb hasn't published any versions yet.")
+	}
+
+	log.Info("\n")
+
+	var (
+		jobs      = 0
+		commands  = 0
+		executors = 0
+	)
+
+	for range info.Orb.Commands {
```

Comment:
`Orb.Commands` is a map, I think you can use `len(info.Orb.Commands)`? (Ditto for executors and jobs).

Reply:
@eric-hu Thanks for the feedback! I've added some of these changes to #191 :bow: